### ISSUE TITLE
Unittest like assert* method required for datadriven.DataDrivenTest

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#223]
PR #211 includes a definition of `assertEqual` in `DataDrivenTest`, as this allows us to write test cases much more conveniently than raw asserts, and we cannot use unittest.TestCase in these tests for technical reasons. It would be useful to define a subset of the unittest `assert*` methods here so we can use them. Say:

- assertTrue, assertFalse
- assertIs, assertIsNot
- assertGreater, assertGreaterEqual
- assertLess, assertLessEqual
- assertRaises

These should be quick to implement, and will suffice for the vast majority of tests we will want to perform in the data driven scenario.

**Comment by dcolligan** [comment]
Here are the methods unittest defines: https://docs.python.org/2/library/unittest.html
```
assertEqual(a, b)	a == b	 
assertNotEqual(a, b)	a != b	 
assertTrue(x)	bool(x) is True	 
assertFalse(x)	bool(x) is False	 
assertIs(a, b)	a is b	2.7
assertIsNot(a, b)	a is not b	2.7
assertIsNone(x)	x is None	2.7
assertIsNotNone(x)	x is not None	2.7
assertIn(a, b)	a in b	2.7
assertNotIn(a, b)	a not in b	2.7
assertIsInstance(a, b)	isinstance(a, b)	2.7
assertNotIsInstance(a, b)	not isinstance(a, b)	2.7

```

**Comment by dcolligan** [comment]
And also (from the same page):
```
assertAlmostEqual(a, b)	round(a-b, 7) == 0	 
assertNotAlmostEqual(a, b)	round(a-b, 7) != 0	 
assertGreater(a, b)	a > b	2.7
assertGreaterEqual(a, b)	a >= b	2.7
assertLess(a, b)	a < b	2.7
assertLessEqual(a, b)	a <= b	2.7
assertRegexpMatches(s, r)	r.search(s)	2.7
assertNotRegexpMatches(s, r)	not r.search(s)	2.7
assertItemsEqual(a, b)	sorted(a) == sorted(b) and works with unhashable objs	2.7
assertDictContainsSubset(a, b)	all the key/value pairs in a exist in b	2.7
```
and
```
assertMultiLineEqual(a, b)	strings	2.7
assertSequenceEqual(a, b)	sequences	2.7
assertListEqual(a, b)	lists	2.7
assertTupleEqual(a, b)	tuples	2.7
assertSetEqual(a, b)	sets or frozensets	2.7
assertDictEqual(a, b)	dicts	2.7
```
and
```
assertRaises(exc, fun, *args, **kwds)	fun(*args, **kwds) raises exc	 
assertRaisesRegexp(exc, r, fun, *args, **kwds)	fun(*args, **kwds) raises exc and the message matches regex r	2.7
```

**Comment by jeromekelleher** [comment]
We don't need all of these. The first set is plenty - if we find a need the others later then we can add them in.

**Comment by dcolligan** [comment]
Fixed in #227
